docs/index.md

```diff
@@ -190,3 +190,246 @@ The last step is to calculate all the specified corrections for the main fake fa
+| `--cut-region` | `string` | The cut to be applied when determining event counts (e.g., `SRlike`, `ARlike`). `process_fractions` always use the `AR_cut`. |
+| `--dry-run` | `bool` | Preview the binning calculations and changes without modifying the config file. |
+
+### Configuration
+
+To enable this feature, you must add a `equipopulated_binning_options` block to the relevant process or correction section in your configuration file. The script uses this block to generate the new binning and will populate the `var_bins` key and update `split_categories` and `split_categories_binedges` if they are left blank i.e. for continuous variables such as `pt_1` (used i.e. in Wjets process).
+
+For example, a manually binned configuration for the `QCD` process:
+
+```yaml
+# Before: Manual binning for QCD process
+target_processes:
+  QCD:
+    split_categories:
+      njets: ["==0", "==1", ">=2"]
+    split_categories_binedges:
+      njets: [-0.5, 0.5, 1.5, 12.5]
+    var_dependence: pt_2
+    var_bins: 
+      "==0": [30.0, 31.6, 33.5, 35.9, 38.9, 43.6, 51.6, 150.0]
+      "==1": [30.0, 32.2, 35.3, 40.3, 50.9, 150.0]
+      ">=2": [30.0, 32.7, 36.5, 42.5, 57.0, 150.0]
+```
+
+Can be replaced with a configuration for automated binning:
+
+```yaml
+# After: Configuration for automated binning
+target_processes:
+  QCD:
+    split_categories:
+      njets: ["==0", "==1", ">=2"]
+    split_categories_binedges:
+      njets: [-0.5, 0.5, 1.5, 12.5]
+    equipopulated_binning_options:
+      variable_config:
+        pt_2:
+          min: 30.0
+          max: 150.0
+          rounding: 2
+      n_bins:  # this part is necessarily only needed for continuous variables
+        njets: 3
+      var_dependence_n_bins: [7, 5, 5]
+    var_dependence: pt_2
+    var_bins: {} # This will be filled by the script
+```
+
+The `equipopulated_binning_options` block has the following structure:
+
+| Parameter | Type | Description |
+|---|---|---|
+| `variable_config` | `dict` | Defines parameters for variables used in binning. See sub-parameters below. |
+| `n_bins` | `dict` | Specifies the desired number of bins per variable in `split_categories` if those are not set there in case of discrete variables, i.e. four equidistant bins in `pt_1`: `pt_1: 4` |
+| `var_dependence_n_bins` |  `int` or `list[int]` or `list[list[int]]` | Number of bins used for the variable defined in `var_dependence`, either as int (used by all categories) or a (nested) list of integers defining number of bins per (nested) category created.|
+
+The `variable_config` for each variable contains:
+
+| Sub-parameter | Type | Description |
+|---|---|---|
+| `min` | `float` | The lower bound for the variable's range. The first bin edge will be set to this value. This cut is applied calculation of equipopulated binning starts.  |
+| `max` | `float` | The upper bound for the variable's range. The last bin edge will be set to this value. This cut is applied calculation of equipopulated binning starts. |
```

Comment:
... This is applied as a cut before the calculation of equipopulated binning starts.